lib/src/cupertino_controls.dart

```diff
@@ -29,28 +29,33 @@ class CupertinoControls extends StatefulWidget {
 
 class _CupertinoControlsState extends State<CupertinoControls>
     with SingleTickerProviderStateMixin {
-  VideoPlayerValue _latestValue;
-  double _latestVolume;
+  late VideoPlayerValue _latestValue;
+  double? _latestVolume;
   bool _hideStuff = true;
-  Timer _hideTimer;
+  Timer? _hideTimer;
   final marginSize = 5.0;
-  Timer _expandCollapseTimer;
-  Timer _initTimer;
+  Timer? _expandCollapseTimer;
+  Timer? _initTimer;
   bool _dragging = false;
 
-  VideoPlayerController controller;
-  ChewieController chewieController;
-  AnimationController playPauseIconAnimationController;
+  late VideoPlayerController controller;
+  ChewieController get chewieController => _chewieController!;
```

Comment:
Why can we not use a late ChewieController here?

Reply:
The problem is the logic in `didChangeDependencies`. This is the only place where `_chewieController` is used directly, and this is required because it tries to compare the old ontroller with the new controller. That would not work with `late` the very first time a controller is set, because there is no old controller and a `LateInitializationError` would be triggered.

The way I wrote this the rest of the widget (especially build methods) don't have to do a null check every time, but `didChangeDependecies` can have access to a nullable `_chewieController`.

Hope that makes sense :)

Reply:
Nice solution then.
Last idea I had was using initState, but then we cannot get the correct context for inheritedWidget.

Could you add a comment for the getter: Something like `// We know that _chewieController is set in didChangeDependencies`

Reply:
Sure, that will make it clearer for readers. Done.